src/vs/workbench/contrib/notebook/common/model/notebookCellTextModel.ts

```diff
@@ -445,24 +446,41 @@ export class NotebookCellTextModel extends Disposable implements ICell {
 			return false;
 		}
 
-		if (this.internalMetadata?.executionOrder !== b.internalMetadata?.executionOrder
-			|| this.internalMetadata?.lastRunSuccess !== b.internalMetadata?.lastRunSuccess
-			|| this.internalMetadata?.runStartTime !== b.internalMetadata?.runStartTime
-			|| this.internalMetadata?.runStartTimeAdjustment !== b.internalMetadata?.runStartTimeAdjustment
-			|| this.internalMetadata?.runEndTime !== b.internalMetadata?.runEndTime) {
-			return false;
+		if (!ignoreMetadata) {
```

Comment:
alternatively, we could handle internal metadata changes more precisely as we do for outputs and the other metadata changes